src/client/driver/index.js

```diff
@@ -140,6 +141,12 @@ export default class ClientDriver {
             });
     }
 
+    _onWaitForElementCommand (command) {
+        executeWaitForElementCommand(command, this.elementAvailabilityTimeout)
+            .catch(err => this._onJsError(err))
```

Comment:
I think waits (like Hybrids) shouldn't be rejected by JS error (BTW is it possible to have JS error here? Or we just handle errors in selector this way currently?).